crates/miden-lib/asm/miden/contracts/auth/multisig.masm

```diff
@@ -1,24 +1,273 @@
 use.miden::account
+use.miden::auth
 use.miden::tx
+
 use.std::crypto::dsa::rpo_falcon512
+use.std::crypto::hashes::rpo
+use.std::sys
 
 # CONSTANTS
 # =================================================================================================
 
-# Event to place the falcon signature of a provided message and public key on the advice stack.
-const.FALCON_SIG_TO_STACK=131087
+# The event to request an authentication signature.
+const.AUTH_REQUEST=131087
+
+# The event emitted when a signature is not found for a required signer.
+const.UNAUTHORIZED_EVENT=131101
+
+# The slot in this component's storage layout where the signature threshold is stored.
+const.THRESHOLD_SLOT=0
+
+# The slot in this component's storage layout where the number of approvers is stored.
+const.NUMBER_OF_APPROVERS_SLOT=1
+
+# The slot in this component's storage layout where the public keys map is stored.
+const.PUBLIC_KEYS_MAP_SLOT=2
 
-# The slot in this component's storage layout where the public key is stored.
-const.PUBLIC_KEY_SLOT=0
+# The slot in this component's storage layout where executed transactions are stored.
+const.EXECUTED_TXS_SLOT=3
 
-    
-#! Authenticate a transaction using the Falcon signature scheme.
-#! This is a single auth procedure that handles both the addition of a signature and the multisig tx execution.
-#! 
+# Runtime Memory Addresses
+# @dev these can be further optimized, (i do prefer to keep the stack clean and incur the cost of calling mem_{store, load})
+const.TEMP_PK=108
+const.MESSAGE=112
+const.SUCCESSFUL_VERIFICATIONS=116
+const.THRESHOLD=120
+
+# ERRORS
+const.NON_CONSTANT_MSG="NON CONSTANT SIGNATURES"
+const.INVALID_SIGNATURE="SIG DOES NOT MATCH DIGEST"
```

Comment:
iirc most of our other error strings are lowercase, I'd stick to that for consistency